### PR TITLE
[FIX] account_credit_control: Allow to use run on multicompany environments

### PR DESCRIPTION
**Pull request by etobella**
With the change, run works properly if no company is set (generated for all visible companies)

### CODE REVIEW

**Comment by etobella** [comment]
@pedrobaeza May I ask for your review?

File: account_credit_control/report/report_credit_control_summary.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
                             <td>
                                 <span t-field="l.date_due" />
                             </td>
-                            <td class="text-right">
+                            <td class="text-righcbt">
```
Comment (pedrobaeza) [inline account_credit_control/report/report_credit_control_summary.xml]: Is this change intended?

File: account_credit_control/models/credit_control_run.py [hunk 2]
```diff
@@ -132,14 +132,15 @@ def _generate_credit_lines(self):
...
                 continue
-            (
-                policy_manual_lines,
-                policy_lines_generated,
-                policy_report,
-            ) = policy._generate_credit_lines(self, {"run_id": self.id})
-            manually_managed_lines |= policy_manual_lines
-            generated |= policy_lines_generated
-            report += policy_report
+            for company in self.company_id or self.env.companies:
```
Comment (pedrobaeza) [inline account_credit_control/models/credit_control_run.py:135]: We shouldn't include in the same run several companies, as their records will be only visible if all companies are allowed at the same time.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_credit_control/report/report_credit_control_summary.xml (hunk 1 above)
Comment (etobella) [reply account_credit_control/report/report_credit_control_summary.xml]: Probably a typo error from my side. Sorry :disappointed:

File: account_credit_control/models/credit_control_run.py (hunk 2 above)
Comment (etobella) [reply account_credit_control/models/credit_control_run.py:135]: Well, right now the field `company_id` on the credit.control.run is not required. Users expects that if the field is null, it will create the credit lines for all the companies visible :thinking: 
We can make it required and forget about making this multicompany. Also we could make the field M2M in order to avoid doubts. What do you prefer?

Comment (pedrobaeza) [reply account_credit_control/models/credit_control_run.py:135]: I think it should be done mandatory. There's no sense of executing a multi-company run that mixes up everything and to have to make the code aware of that. Imagine having the same partner with debts in several companies.

Unless it's your use case that you want to cover in one shot...

Comment (etobella) [reply account_credit_control/models/credit_control_run.py:135]: The idea was to generate all the lines on the same execution

But, data is not mixed, communications and credit lines are created by company and partner, not only by partner, so they are not mixed.

Comment (pedrobaeza) [reply account_credit_control/models/credit_control_run.py:135]: They all belong to the same run, which is the inconsistent thing.

Comment (etobella) [reply account_credit_control/models/credit_control_run.py:135]: It is not if it has no company or multiple company associated

Comment (pedrobaeza) [reply account_credit_control/models/credit_control_run.py:135]: Yes. The problem is the or party. Make required the field company_id
